File: ios/Sources/Drengr/NetworkEvent.swift
import Foundation

public struct NetworkEvent {
    public let method: String
    public let url: String
    public let statusCode: Int?
    public let durationMs: Int
    public let requestBodyBytes: Int
    public let responseBodyBytes: Int
    public let requestHeaders: [String: String]
    public let responseHeaders: [String: String]
    public let requestBody: String?
    public let responseBody: String?
    public let errorText: String?
    public let timestampMs: Int64
}

File: android/drengr/src/main/kotlin/dev/drengr/sdk/NetworkEvent.kt
package dev.drengr.sdk

/** A single captured HTTP exchange (already redacted by the capture layer). */
data class NetworkEvent(
    val method: String,
    val url: String,
    val statusCode: Int?,
    val durationMs: Long,
    val requestBodyBytes: Long,
    val responseBodyBytes: Long,
    val requestHeaders: Map<String, String>,
    val responseHeaders: Map<String, String>,
    /** Redacted request body text (textual bodies only, capped). */
    val requestBody: String?,
    /** Redacted response body text (textual bodies only, capped). */
    val responseBody: String?,
    val errorText: String?,
    val timestampMs: Long,
)
